r1cs/gadgets/std/src/groups/nonnative/nonnative_group_gadget.rs

```diff
@@ -284,6 +282,104 @@ for GroupAffineNonNativeGadget<P, ConstraintF, SimulationF>
         result = result.sub_constant(cs.ns(|| "result - sigma*n_div_2"), &to_sub)?;
         Ok(result)
     }
+
+    fn get_value(&self) -> Option<<Self as GroupGadget<SWProjective<P>, ConstraintF>>::Value> { 
+        unimplemented!()
+    }
+
+    fn get_variable(&self) -> Self::Variable {
+        unimplemented!()
+    }
+
+    fn cost_of_add() -> usize {
+        unimplemented!()
+    }
+
+    fn cost_of_double() -> usize {
+        unimplemented!()
+    }
+
+    fn sub<CS: ConstraintSystem<ConstraintF>>(
```

Comment:
All the functions below have default implementation in the GroupGadget trait, so you can also remove them here.ù (you just copy pasted their implementation anyway)

Reply:
Done